In [5]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

In [6]:
consumo_y = [50, 52, 55, 53, 56, 58]
ingreso_x1 = [80, 82, 85, 83, 88, 90]
inflacion_x2 = [4.0, 4.5, 4.0, 5.0, 4.2, 4.0]
tasa_x3 = [6.0, 6.5, 6.0, 7.0, 6.2, 6.0]

In [9]:
x_n = np.stack((ingreso_x1, inflacion_x2, tasa_x3))
model = LinearRegression()
model.fit(x_n.T, consumo_y)
print("Intercepto:", model.intercept_)
print("Coeficientes:", model.coef_)
print('Consumo =', model.intercept_, '+', model.coef_[0], 'Ingreso +', model.coef_[1], 'Inflacion +', model.coef_[2], 'Tasa')

Intercepto: -11.923523779125418
Coeficientes: [0.76444303 0.11362911 0.11362911]
Consumo = -11.923523779125418 + 0.7644430258538141 Ingreso + 0.11362910947973173 Inflacion + 0.1136291094797317 Tasa


## ¿Qué variable explica mejor el consumo?
En este caso es el ingreso porque es el coeficiente con el valor mas alto, positivamente.

## Interpreta β2​ en términos económicos.
La inflacion tiene un impacto en el consumo, por cada unidad se aumenta 0.11 en el consumo

## ¿Qué problema habría si se omite la inflación?
Seria mas dificil interpretar y predecir el consumo